app/models/vendor.rb

```diff
@@ -40,7 +40,7 @@ def self.accessible_by(user)
   # calls on product, product tests, etc, all the way down and does them manually. This means that
   # if any of those structures change then this code will need to be updated accordingly.
   def destroy
-    product_tests = ProductTest.where(:product_id.in => product_ids)
+    product_tests = ProductTest.where(:product_id.in => product_ids).by_updated_at
```

Comment:
Can we review the code for commonly used where conditions and make them into scopes? This doesn't have to be done as part of this PR, but it's something to look at.